gutenberg.php

```diff
@@ -16,20 +16,144 @@
+function gutenberg_version_too_old_notice() {
+	echo '<div class="error"><p>';
+
+	if ( current_user_can( 'update_plugins' ) ) {
+		printf(
+			/* translators: %s: URL to the Plugins screen */
+			__( 'The Gutenberg plugin cannot be used. It is too old for your version of WordPress. Please <a href="%s">turn auto-updates on</a> or updage it manually.', 'gutenberg' ),
+			admin_url( 'plugins.php' )
+		);
+	} else {
+		_e( 'The Gutenberg plugin cannot be used. It is too old for your version of WordPress. Please ask an administratior to updage it.', 'gutenberg' );
+	}
+
+	echo '</p></div>';
+}
+
+/**
+ * Add a "WordPress version is too old" plugins list table notice.
+ *
+ * @since 11.7.0
+ *
+ * @param string[] $links Array of plugin action links.
+ * @param string   $file  Path to the plugin file relative to the plugins directory.
+ * @return string[] Updated array of plugin action links.
+ */
+function gutenberg_wordpress_version_too_old_action_links_notice( $links, $file ) {
+	$plugin_basename = basename( __DIR__ ) . '/gutenberg.php';
+
+	if ( $file === $plugin_basename && current_user_can( 'manage_options' ) ) {
+		// Prevent PHP warnings when a plugin uses this filter incorrectly.
+		$links = (array) $links;
+
+		// Can only be a very short notice.
+		$text = __( 'WordPress is too old. Gutenberg cannot be used.', 'gutenberg' );
+
+		// The 'button-link-delete' class name makes the text red.
+		$links['wp-gutenberg-plugin-notice button-link-delete'] = esc_html( $text );
+	}
+
+	return $links;
+}
+
+/**
+ * Add a "Gutenberg version is too old" plugins list table notice.
+ *
+ * @since 11.7.0
+ *
+ * @param string[] $links Array of plugin action links.
+ * @param string   $file  Path to the plugin file relative to the plugins directory.
+ * @return string[] Updated array of plugin action links.
+ */
+function gutenberg_version_too_old_action_links_notice( $links, $file ) {
+	$plugin_basename = basename( __DIR__ ) . '/gutenberg.php';
+
+	if ( $file === $plugin_basename && current_user_can( 'manage_options' ) ) {
+		// Prevent PHP warnings when a plugin uses this filter incorrectly.
+		$links = (array) $links;
+
+		// Can only be a very short notice.
+		$text = __( 'Gutenberg is too old and cannot be used.', 'gutenberg' );
+
+		// The 'button-link-delete' class name makes the text red.
+		$links['wp-gutenberg-plugin-notice button-link-delete'] = esc_html( $text );
+	}
 
-	deactivate_plugins( array( 'gutenberg/gutenberg.php' ) );
+	return $links;
+}
+
+/**
+ * Add a "Gutenberg version is getting old. Please update!" notice on the Dashboard.
+ *
+ * @since 11.7.0
+ */
+function gutenberg_needs_update_notice() {
+	echo '<div class="notice notice-warning is-dismissible"><p>';
+
+	printf(
+		/* translators: %s: Too new, unsupported WordPress version */
+		__( 'Your version of Gutenberg is getting too old. It will stop working after you upgrade WordPress to %s. Please turn auto-updates on for the Gutenberg plugin or updage it manually.', 'gutenberg' ),
```

Comment:
```suggestion
		__( 'Your version of Gutenberg is getting too old. It will stop working after you upgrade WordPress to %s. Please turn auto-updates on for the Gutenberg plugin or update it manually.', 'gutenberg' ),
```